spec/requests/groups_controller_spec.rb

```diff
@@ -0,0 +1,108 @@
+require 'rails_helper'
+
+RSpec.describe 'Groups', type: :request do
```

Comment:
These controller tests are great  - well-written and thorough. 👍 

My personal preference is to write "fewer integration tests" over "more controller tests" for reasons indicated [here](https://robots.thoughtbot.com/how-we-test-rails-applications#controller-specs) and [here](https://semaphoreci.com/blog/2014/02/11/rails-testing-antipatterns-controllers.html). I make an exception when the controllers have tons of logic.

This is a philosophical discussion, however, and I'm happy to have it elsewhere - I don't want to hold this PR up.